src/components/DatePicker.vue

```diff
@@ -0,0 +1,113 @@
+<!-- ADD_LICENSE_HEADER -->
+<template>
+    <div class="input-group">
+        <div class="input-group-prepend">
+            <span class="input-group-text">Date</span>
+        </div>
+        <input
+            :key="componentKey"
+            v-bind:value="dateString"
+            v-on:input="validateDate"
+            v-on:change="submitChange"
+            placeholder="dd.MM.yyyy"
+            class="form-control" />
+    </div>
+</template>
+
+<script>
+
+export default {
+	name: 'datepicker',
+	components: {
+    },
+    model: {
+        prop: 'value',
+        event: 'input'
+    },
+    props: {
+        format: String,
+        value: String,
+    },
+	data: function () {
+        return {
+            internal_value: null,
+            initial_value: null,
+            componentKey: 0,
+            isInitializing: true,
+            dateRegexp: /^((([1-9]|0[1-9]|[12]\d|3[01])\.([1-9]|0[1-9]|1[0-2])(\.[12]\d{3})?)|([1-9]|0[1-9]|[12]\d|3[01]))$/
```

Comment:
Please comment this so that people who are bad with regex (like me..) can get the logic

Reply:
I will write a unit tests for all these in the next PR.